geographic_msgs/CMakeLists.txt

```diff
@@ -1,9 +1,8 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.14)
```

Comment:
For humble. 
![image](https://github.com/ros-geographic-info/geographic_info/assets/25047695/eb770b56-0060-4dd7-bb23-dfee110fa603)

Reply:
If we don't use 3.13, then the nasty logic about linking the message cpp library to its interface has to be brought up to the top level. 
https://cmake.org/cmake/help/latest/policy/CMP0079.html#cmp0079

I would prefer to abide by REP2000 and set the minimum version to 3.14.

Reply:
I've downgraded to 3.5 and moved all the logic to the top level CMakeLists.

Reply:
FYI @SteveMacenski 

I tried installing CMake 3.5. It is no longer possible to compile any message packages anymore. ament_cmake_core requires 3.12 on rolling nowadays. 

CMake 3.5 was released 7 years ago. 
CMake 3.12 was released 5 years ago. 
Both were released well before humble. 
CMake 3.5 was released before C++17 was out, so it does not support the CXX_STANDARD of 17.

All I ask that you please be a little more mindful of your requests on PR's considering this request to downgrade back to 3.5 is
* Against REP-2000
* Wasting my time as a contributor

I wouldn't have chucked an untested PR at you if I didn't know it worked on `humble`. 

Sure, ros2/common_interfaces is on 3.5, but a little trust that I know what I'm doing would be nice.

Reply:
My issue wasn’t with 3.12 or any particular version, my issue is that the original PR changed many unrelated things to the listed change that you wanted: to add the validation method. 

If you want to change how the cmake is written or versions, that can be done in another PR but mixing topics together is not good process and makes it harder to review. I have to scrutinize every change with respect to whether its a ‘nice to have’ refactor or update as a necessity to add in a util function since they’re presented together.  I need to untangle proof that they’re unrelated, I can’t know for sure on a quick glance. It’s far easier to revert to minimal changes needed for the described update of the PR. 

I appreciate that you tested it, but you didn’t specify what you tested in the PR body and I often get PRs that don’t work (sometimes don’t build) so I can’t read your mind about what you did or did not test and in what environment. Look at it from a reviewers perspective - I only see what you wrote initially and a PR updating a lot of cmake that I can’t decouple from just adding what seems to be a usual nothing-crazy-to-see-here util. 

If you want to update cmake versions or refactor the cmake itself to use modern syntax, we can do that, but PRs as discrete modules are far easier to review and test. 5 small PRs get merged 2x faster than 1 big PR.